config/agents/opencode_web.py

```
@@ -0,0 +1,810 @@
+def validate_neurodesk_api_key(key, base_url):
+    """Validate a candidate key against the LiteLLM /models endpoint.
+
+    Mirrors the terminal wrapper's acceptance rules: 200 and 404 (model list
+    hidden) count as verified, 401/403 is a rejected key, and an unreachable
+    endpoint accepts the key with a warning rather than locking the user out.
+    Returns (accepted, verified, message).
+    """
+    request = urllib.request.Request(
+        f"{base_url.rstrip('/')}/models",
+        headers={"Authorization": f"Bearer {key}"},
+    )
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)